platform/ios/src/MGLAnnotationView.h

```diff
@@ -89,7 +89,7 @@ typedef NS_ENUM(NSUInteger, MGLAnnotationViewDragState) {
  contains a non-`nil` value while the annotation view is visible on the map. If
  the view is queued, waiting to be reused, the value is `nil`.
  */
-@property (nonatomic, readonly, nullable) id <MGLAnnotation> annotation;
+@property (nonatomic, nullable) id <MGLAnnotation> annotation;
```

Comment:
Changing the annotation in `mapView:viewForAnnotation:` will not affect `annotationViewForAnnotation:` (the caller) it's mean to be a convenient way to set up any UI update code. This is also how Mapkit behaves.
From a conversation with @boundsj 
> The developer setting it in the delegate first is redundant but facilitates them writing code to update the view when the annotation model property is set.